src/RMPI/onesided.jl

```diff
@@ -137,7 +138,24 @@ function sbuffer(s::MPIOneSided) # safe version for reading shared buffer - retu
     return res
 end
 
-function Rimu.sort_into_targets!(target, bufs::Vector{Vector{P}}, lens, ::Type{P}, s::MPIOneSided) where P
+function Rimu.sort_into_targets!(target, source, s::MPIOneSided{P}) where P
+    # now target is just a local data structure, e.g. DVec
+    # allocate local buffers for sorting
+    bufs = [Vector{P}(undef,length(source)) for i in 1:(s.np-1)] # type-stable
+    lens = zeros(Int,(s.np-1))
+    # sort source into send buffers
+    @inbounds for (key,val) in pairs(source)
```

Comment:
```suggestion
    @inbounds for (key,val) in pairs(storage(source))
```
?

Reply:
Fixed this by extracting the `source` at the point when `sort_into_targets` gets called on `MPIData`.